### ISSUE TITLE
Silent crash after ~2 months

### ISSUE THREAD

**Issue by Bastian** [Javacord/Javacord#274]
After ~2 month uptime, my bot just crashed silently.
The log file doesn't show any crash, it just stopped logging anything and the bot went offline.
This are the last few log messages before the bot went offline:
```
[DEBUG][ReadingThread][org.javacord.util.gateway.DiscordWebSocketAdapter] Heartbeat ACK received
[DEBUG][Javacord - Central Scheduler - 1][org.javacord.util.gateway.DiscordWebSocketAdapter] Sent heartbeat (interval: 41250)
[DEBUG][ReadingThread][org.javacord.util.gateway.DiscordWebSocketAdapter] Heartbeat ACK received
[DEBUG][Javacord - Central Scheduler - 1][org.javacord.util.gateway.DiscordWebSocketAdapter] Sent heartbeat (interval: 41250)
[DEBUG][ReadingThread][org.javacord.util.gateway.DiscordWebSocketAdapter] Heartbeat ACK received
[DEBUG][Javacord - Central Scheduler - 1][org.javacord.util.gateway.DiscordWebSocketAdapter] Sent heartbeat (interval: 41250)
```
The Java application itself was still running:
![](https://i.imgur.com/xZ7Qe2i.png)

As the bot was online for quite a while, I'm not sure if the bug might already be fixed.

**Comment by Bastian** [comment]
Closing this as long as there are no new reports of this bug.

**Comment by Bastian** [comment]
After some research and a bug-report from @WalkerKnapp, it seems like this is a bug from our websocket library keeping stuck in `CLOSING` state.
If you compare the correct behavior after a missed heartbeat ack:
```xml
2018-12-28 15:00:04,061 <TRACE> <WritingThread> <[]> <{shard=0}> <org.javacord.core.util.logging.WebSocketLogger> onSendingFrame: frame='WebSocketFrame(FIN=1,RSV1=0,RSV2=0,RSV3=0,Opcode=CLOSE,Length=37,CloseCode=4000,Reason="Heartbeat was not answered properly")'
2018-12-28 15:00:04,062 <TRACE> <WritingThread> <[]> <{shard=0}> <org.javacord.core.util.logging.WebSocketLogger> onStateChanged: newState='CLOSING'
2018-12-28 15:00:04,062 <TRACE> <WritingThread> <[]> <{shard=0}> <org.javacord.core.util.logging.WebSocketLogger> onFrameSent: frame='WebSocketFrame(FIN=1,RSV1=0,RSV2=0,RSV3=0,Opcode=CLOSE,Length=37,CloseCode=4000,Reason="Heartbeat was not answered properly")'
2018-12-28 15:00:04,200 <TRACE> <ReadingThread> <[]> <{shard=0}> <org.javacord.core.util.logging.WebSocketLogger> onFrame: frame='WebSocketFrame(FIN=1,RSV1=0,RSV2=0,RSV3=0,Opcode=CLOSE,Length=2,CloseCode=4000,Reason=null)'
2018-12-28 15:00:04,200 <TRACE> <ReadingThread> <[]> <{shard=0}> <org.javacord.core.util.logging.WebSocketLogger> onCloseFrame: frame='WebSocketFrame(FIN=1,RSV1=0,RSV2=0,RSV3=0,Opcode=CLOSE,Length=2,CloseCode=4000,Reason=null)'
2018-12-28 15:00:04,201 <TRACE> <ReadingThread> <[]> <{shard=0}> <org.javacord.core.util.logging.WebSocketLogger> onStateChanged: newState='CLOSED'
2018-12-28 15:00:04,202 <INFO> <ReadingThread> <[]> <{shard=0}> <org.javacord.core.util.gateway.DiscordWebSocketAdapter> Websocket closed with reason 'Heartbeat was not answered properly' and code UNKNOWN_ERROR (4000) by client!
2018-12-28 15:00:04,203 <INFO> <ReadingThread> <[]> <{shard=0}> <org.javacord.core.util.gateway.DiscordWebSocketAdapter> Trying to reconnect/resume in 1 seconds!
```
with a "silent crash", you can clearly see that it never changed from `CLOSING` to `CLOSED` state:
```xml
07:22:51.261 [WritingThread] TRACE org.javacord.core.util.logging.WebSocketLogger - onSendingFrame: frame='WebSocketFrame(FIN=1,RSV1=0,RSV2=0,RSV3=0,Opcode=CLOSE,Length=37,CloseCode=4000,Reason="Heartbeat was not answered properly")'
07:22:51.261 [WritingThread] TRACE org.javacord.core.util.logging.WebSocketLogger - onStateChanged: newState='CLOSING'
07:22:51.262 [WritingThread] TRACE org.javacord.core.util.logging.WebSocketLogger - onFrameSent: frame='WebSocketFrame(FIN=1,RSV1=0,RSV2=0,RSV3=0,Opcode=CLOSE,Length=37,CloseCode=4000,Reason="Heartbeat was not answered properly")'
```

At the moment I'm trying to reproduce this bug.

**Comment by Bastian** [comment]
I can now confirm, that this bug still exists in the latest version:
![](https://i.imgur.com/TXYmxf5.png)
This happend after some Discord itself had some issues.

**Comment by Bastian** [comment]
It's still possible, that this is already fixed in my audio PR (#396) which updates the websocket dependency from `1.31` to `2.6`.

**Comment by Bastian** [comment]
There were no more incidents since 3.0.2, so I assume that this is fixed now.
